File: solutions/javascript/collatz-conjecture/1/collatz-conjecture.js
//
// This is only a SKELETON file for the 'Collatz Conjecture' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const steps = (num = 1) => {
  if (num <= 0) {
    throw new Error('Only positive integers are allowed');
  }
  let step = 0;
  while (num !== 1) {
    if ( num % 2 === 0) {
      num /= 2;
    }else {
      num = num * 3 + 1 ;
    }
    step++;
  }
  return step;
};
 

File: solutions/ocaml/collatz-conjecture/1/collatz_conjecture.ml
let rec collatz_steps i n =
  if n <= 0 then Result.error "Only positive integers are allowed"
  else if n = 1 then Result.ok i
  else if n mod 2 = 0 then collatz_steps (i + 1) (n / 2)
  else collatz_steps (i + 1) (3 * n + 1)
let collatz_conjecture = collatz_steps 0
